tests/retrievers/test_embeddings.py

```diff
@@ -0,0 +1,85 @@
+import numpy as np
+import pytest
+
+from dspy.retrievers.embeddings import Embeddings
+
+@pytest.fixture
+def dummy_corpus():
+    return [
+        "The cat sat on the mat.",
+        "The dog barked at the mailman.",
+        "The sun rises in the east.",
+        "The quick brown fox jumps over the lazy dog.",
+        "An apple a day keeps the doctor away."
+    ]
+
+@pytest.fixture
+def dummy_embedder():
+    def embed(texts):
+        # Simple dummy embedder: convert each character's ASCII values and sum for each string
+        embeddings = []
+        for text in texts:
+            vec = np.array([sum(ord(c) for c in text[i::10]) for i in range(10)], dtype=np.float32)
+            embeddings.append(vec)
+        return np.stack(embeddings)
+    return embed
+
```

Comment:
Let's add a test case to test calling the embedding at multithreading setup. Behind the scene we are queuing the embedding queries to batch then call embedding models in batch, which could have issues (low chance) on high concurrency.